cmd/mistryd/metrics_recorder.go

```diff
@@ -0,0 +1,124 @@
+package main
```

Comment:
I think this should go into its own package, named `metrics` or `metrics_recorder`, as opposed to package `main`. Then the API would look like this:

```go
metrics.New() // or metrics_recorder.New()
// etc.
```

Note that it doesn't have to be in `<root>/pkg`, it can be located at `<root>/cmd/mistryd/metrics/`.